File: src/main/java/io/vproxy/vfx/ui/pane/TransparentFusionPane.java
package io.vproxy.vfx.ui.pane;

import io.vproxy.vfx.theme.Theme;
import javafx.scene.paint.Color;

public class TransparentFusionPane extends FusionPane {
    public TransparentFusionPane() {
        super();
    }

    public TransparentFusionPane(boolean manuallyHandleOuterRegion) {
        super(manuallyHandleOuterRegion);
    }

    @Override
    protected AbstractFusionPane buildRootNode() {
        return new TransparentFusionPaneImpl();
    }

    protected class TransparentFusionPaneImpl extends FusionPaneImpl {
        @Override
        protected Color normalColor() {
            return Theme.current().transparentFusionPaneNormalBackgroundColor();
        }

        @Override
        protected Color hoverColor() {
            return Theme.current().transparentFusionPaneHoverBackgroundColor();
        }
    }
}

File: src-ts/src/ui/pane/TransparentFusionPane.ts
import { Color } from '../../javafx/color.js';
import { Theme } from '../../theme/Theme.js';
import { FusionPane } from './FusionPane.js';
import { FusionPaneImpl } from './FusionPane.js';

export class TransparentFusionPane extends FusionPane {
  constructor();
  constructor(manuallyHandleOuterRegion: boolean);
  constructor(manuallyHandleOuterRegion?: boolean) {
    super(manuallyHandleOuterRegion ?? true);
  }

  protected override buildRootNode() {
    return new TransparentFusionPaneImpl(this);
  }
}

export class TransparentFusionPaneImpl extends FusionPaneImpl {
  protected override normalColor(): Color {
    return Theme.current().transparentFusionPaneNormalBackgroundColor();
  }

  protected override hoverColor(): Color {
    return Theme.current().transparentFusionPaneHoverBackgroundColor();
  }
}
